functions/HttpTriggerAPIUsersId/__init__.py

```diff
@@ -58,8 +85,107 @@ def get_user_by_id(conn, id):
             columns = [column[0] for column in cursor.description]
             data = dict(zip(columns, row))
 
+            logging.debug("Users retrieved successfully!")
             return func.HttpResponse(
                 json.dumps(data),
                 status_code=200,
                 mimetype="application/json"
             )
+
+
+def post():
+    logging.debug("POST is not implemented")
+    return func.HttpResponse(
+        "Method not allowed!",
+        status_code=405
+    )
+
+
+def put(req, conn, id):
+    user_req_body = req.get_json()
+    logging.debug("Verifying fields in request body to update a user by ID")
+    try:
+        assert "firstName" in user_req_body, "User request body did not contain field: 'firstName'"
+        assert "lastName" in user_req_body, "User request body did not contain field: 'lastName'"
+        assert "email" in user_req_body, "User request body did not contain field: 'email'"
+    except AssertionError as user_req_body_content_error:
+        logging.error(
+            "User request body did not contain the necessary fields!")
+        return func.HttpResponse(user_req_body_content_error.args[0], status_code=400)
+
+    logging.debug("User request body contains all the necessary fields!")
+    with conn.cursor() as cursor:
+        # Unpack user data
+        firstName = user_req_body["firstName"]
+        lastName = user_req_body["lastName"]
+        email = user_req_body["email"]
+        user_params = (firstName, lastName, email)
+
+        # Update user  query
+        update_user_query = "UPDATE dbo.users SET firstName = ?, lastName = ?, email = ? WHERE userId={}".format(
+            id)
+
+        try:
+            logging.debug("Check if userId exists: " + id)
+            cursor.execute(
+                "SELECT * FROM users WHERE userId={}".format(id))
+            row = cursor.fetchone()
+            if not row:
+                return func.HttpResponse(
+                    "User not found",
+                    status_code=404
+                )
+            logging.debug("Executing query: " + update_user_query)
+            cursor.execute(update_user_query, user_params)
+            logging.debug("User was updated successfully!.")
+            return func.HttpResponse(
+                "User updated",
+                status_code=200
+            )
+        except Exception as e:
```

Comment:
It would be nice if this was a more specific exception rather than just Exception. Though, to be fair, I think I also used a generic exception for this failing point as well.

Reply:
I have not included any specific error handling. This can be updated later if needed